### PR TITLE
Integrate sync stack in schema registry via AutoRest.

### PR DESCRIPTION
**Pull request by g2vinay**
Integrates Sync Stack in Schema Registry via AutoRest, also enables ResponseBase in Schema Registry via AutoRest.

### CODE REVIEW

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java [hunk 1]
```diff
@@ -290,8 +290,8 @@ Mono<Response<SchemaRegistrySchema>> getSchemaWithResponse(String groupName, Str
                 context)
             .onErrorMap(ErrorException.class, SchemaRegistryAsyncClient::remapError)
             .handle((response, sink) -> {
-                final InputStream schemaInputStream = response.getValue();
-                final SchemaProperties schemaObject = SchemaRegistryHelper.getSchemaProperties(response);
+                final InputStream schemaInputStream = response.getValue().toStream();
```
Comment (srnagar) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java:293]: Why do we use `InputStream` in async client? Now that we have `BinaryData`, can we use `toFlux()` instead?

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClient.java [hunk 2]
```diff
@@ -122,8 +143,27 @@ public SchemaProperties registerSchema(String groupName, String name, String sch
...
+        } else if (Objects.isNull(schemaDefinition)) {
+            throw logger.logExceptionAsError(new NullPointerException("'schemaDefinition' should not be null."));
+        } else if (Objects.isNull(format)) {
+            throw logger.logExceptionAsError(new NullPointerException("'format' should not be null."));
+        }
+
+        logger.verbose("Registering schema. Group: '{}', name: '{}', serialization type: '{}', payload: '{}'",
+            groupName, name, format, schemaDefinition);
+
+        context = enableSyncRestProxy(context);
```
Comment (srnagar) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClient.java:159]: Do we need to set this for every method? Can this be changed to a policy instead and set at the client level?

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java [hunk 3]
```diff
@@ -53,20 +54,33 @@ public final class SchemasGetByIdHeaders {
...
-        this.schemaId = rawHeaders.getValue("Schema-Id");
-        this.schemaGroupName = rawHeaders.getValue("Schema-Group-Name");
-        this.schemaName = rawHeaders.getValue("Schema-Name");
-        this.schemaIdLocation = rawHeaders.getValue("Schema-Id-Location");
-        this.location = rawHeaders.getValue("Location");
-        this.contentType = rawHeaders.getValue("Content-Type");
+        String schemaVersion = rawHeaders.getValue(SCHEMA_VERSION);
+        if (schemaVersion != null) {
+            this.schemaVersion = Integer.parseInt(schemaVersion);
+        }
```
Comment (srnagar) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java:77]: Previously, we didn't check for null and if the schema version was null, we'd throw an exception. This will change the behavior.

Comment (g2vinay) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java:77]: This seems to be changed by Autorest generation.
We can hand revert it with a note here. 
Open to other ideas.

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClient.java (hunk 2 above)
Comment (g2vinay) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClient.java:159]: Since it applies to all sync APIs in this case, that's doable I believe.

Comment (g2vinay) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClient.java:159]: This cannot be done at policy level, as this info is read at Rest Proxy level through the Context. 
Policies come into play post Rest Proxy I believe.
Also, this change is temporary and will be removed later on once we enable sync stack by default across all SDKs.

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java (hunk 1 above)
Comment (g2vinay) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java]: InputStream was being used before, so didn't intend to change the logic as part of sync stack PR.
I've updated to use Flux<ByteBuffer> here, do verify if it looks good.

**Review by srnagar — APPROVED** [review]

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/SchemaRegistryHelper.java [hunk 4]
```diff
@@ -38,29 +41,29 @@ public static void setAccessor(SchemaRegistryModelsAccessor modelsAccessor) {
         accessor = Objects.requireNonNull(modelsAccessor, "'modelsAccessor' cannot be null.");
     }
 
-    public static SchemaProperties getSchemaProperties(SchemasRegisterResponse response) {
-        final SchemasRegisterHeaders headers = response.getDeserializedHeaders();
+    public static SchemaProperties getSchemaPropertiesFromSchemasGetSchemaVersionHeaders(ResponseBase<SchemasGetSchemaVersionHeaders, BinaryData> response) {
```
Comment (conniey) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/SchemaRegistryHelper.java:44]: Instead of creating these differently named overloads, can we use the same method name but take headers instead of the whole response?
```java
public static SchemaProperties getSchemaProperties(SchemasGetSchemaVersionHeaders headers) {
}

public static SchemaProperties getSchemaProperties(SchemasQueryIdByContentHeaders headers) {
}
```

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClientBuilder.java [hunk 5]
```diff
@@ -449,6 +454,7 @@ public SchemaRegistryAsyncClient buildAsyncClient() {
      * and {@link #retryPolicy(RetryPolicy)} have been set.
      */
     public SchemaRegistryClient buildClient() {
-        return new SchemaRegistryClient(this.buildAsyncClient());
+        AzureSchemaRegistryImpl restService = getAzureSchemaREgistryImplService();
```
Comment (conniey) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryClientBuilder.java:457]: nit: capital RE rather than Re

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java [hunk 6]
```diff
@@ -421,45 +398,34 @@ Mono<Response<SchemaProperties>> getSchemaPropertiesWithResponse(String groupNam
...
      *
-     * @param inputStream Input stream.
+     * @param byteBufferFlux the Byte Buffer Flux input.
      *
      * @return A string representation.
      *
-     * @throws UncheckedIOException if an {@link IOException} is thrown when creating the readers.
      */
-    private static String convertToString(InputStream inputStream) {
+    static Mono<String> convertToString(Flux<ByteBuffer> byteBufferFlux) {
```
Comment (conniey) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/SchemaRegistryAsyncClient.java:423]: I remember manually changing it to InputStream for a reason... I think it was trying to deserialize or reserialize the payload... but if this works in the integration tests then yay!

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java [hunk 7]
```diff
@@ -197,7 +211,7 @@ public SchemasGetByIdHeaders setLocation(String location) {
      *
      * @return the contentType value.
      */
-    public SchemaFormatImpl getContentType() {
+    public SchemaFormat getContentType() {
```
Comment (conniey) [inline sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java:214]: Curious why the change from SchemaFormatImpl -> SchemaFormat? I think there is a collision with the public-facing one so we have to qualify it. :/

**Review by conniey — APPROVED** [review]

File: sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java (hunk 7 above)
Comment (g2vinay) [reply sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/models/SchemasGetByIdHeaders.java:214]: This is done by Autorest, we can revert this if its a breaking change ?
